Source/Core/VideoBackends/Software/Tev.cpp

```diff
@@ -369,25 +369,34 @@ static bool TevAlphaTest(int alpha)
 
 static inline s32 WrapIndirectCoord(s32 coord, int wrapMode)
 {
+	s32 size;
+
 	switch (wrapMode)
 	{
-		case ITW_OFF:
-			return coord;
 		case ITW_256:
-			return (coord % (256 << 7));
+			size = (256 << 7);
+			break;
 		case ITW_128:
-			return (coord % (128 << 7));
+			size = (128 << 7);
+			break;
 		case ITW_64:
-			return (coord % (64 << 7));
+			size = (64 << 7);
+			break;
 		case ITW_32:
-			return (coord % (32 << 7));
+			size = (32 << 7);
+			break;
 		case ITW_16:
-			return (coord % (16 << 7));
+			size = (16 << 7);
+			break;
 		case ITW_0:
 			return 0;
+		case ITW_OFF:
+			return coord;
 		default:
 			return 0;
 	}
+
+	return (coord < 0) ? (size - (std::abs(coord) % size)) : (std::abs(coord) % size);
```

Comment:
@sepalani Yes, fair point. Won't work in the pixel shader though, due to the % operator being undefined if sign(x) is not the same as sign(y) for GLSL, and it seemed the "broken" drivers were returning negative numbers, whereas the "working" drivers were always returning positive.